### ISSUE TITLE
RPC API log messages go to STDERR, not to log file

### ISSUE THREAD

**Issue by matsduf** [zonemaster/zonemaster-backend#1013]
When starting RPC API log messages go to STDERR:
```
$ /usr/local/etc/rc.d/zm_rpcapi start
Starting zm_rpcapi.
2022-05-18T07:01:27Z [61426] [NOTICE] [Zonemaster::Backend::Config] Loading config: /usr/local/etc/zonemaster/backend_config.ini
2022-05-18T07:01:27Z [61426] [INFO] [Plack::Sandbox::_2fusr_2flocal_2fbin_2fzonemaster_backend_rpcapi_2epsgi] Enabling add_batch_job method
```

When running a test, log messages go to STDERR where the daemon was started (*)
```
[root@fbsd13-be1 ~]# 2022-05-18T07:02:59Z [61429] [NOTICE] [Zonemaster::Backend::DB] Connecting to database 'DBI:SQLite:dbname=/var/db/zonemaster/db.sqlite' Extra parameters: {'rpc_method' => 'start_domain_test'}
```

The messages do not go to the log file, where they are expected:
```
$  cat /var/log/zonemaster/zm_rpcapi.log
Process Backgrounded
2022/05/18-06:46:38 Starman::Server (type Net::Server::PreFork) starting! pid(61154)
Binding to TCP port 5000 on host 127.0.0.1 with IPv4
2022/05/18-07:00:48 Server closing!
Process Backgrounded
2022/05/18-07:01:16 Starman::Server (type Net::Server::PreFork) starting! pid(61401)
Binding to TCP port 5000 on host 127.0.0.1 with IPv4
2022/05/18-07:01:21 Server closing!
Process Backgrounded
2022/05/18-07:01:27 Starman::Server (type Net::Server::PreFork) starting! pid(61428)
Binding to TCP port 5000 on host 127.0.0.1 with IPv4
```

(*) Only for the first test logs are printed to STDERR. After that no logs are created.

Installed from develop branch on commit https://github.com/zonemaster/zonemaster-backend/commit/f82734cf9993bca0350dc7ce06a7461520a58507.

**Comment by blacksponge** [comment]
I managed to reproduce this issue successfully. From what I see I also have this issue on the master branch, can you confirm?

**Comment by matsduf** [comment]
Previously no log messages in that format were outputted at start of the daemon, and especially no log message at all was outputted when running the first test.

When the lack of messages to the log file has not changed.

Something has changed, new structured messages are created. I expected those to appear in the log file.

**Comment by blacksponge** [comment]
Okay I think I understand what's going on. First in all cases the pre-fork log messages are sent to stderr, that would be the two that you have when starting the daemon.

On master following messages are then redirected to the log file while on develop there are still displayed on stderr. The difference I see is that on develop we are duplicating stderr, bypassing the redirection that is later done.

I introduced the stderr / stdout duplication because that was the way it was done in the test agent.
@mattias-p you were the one introducing this behaviour, do you think we avoid stderr / stdout duplication and use them directly or was there a specific reason to do this?

**Comment by blacksponge** [comment]
Decided during meeting to not duplicate STDERR.  
PR https://github.com/zonemaster/zonemaster-backend/issues/1013 implements the fix.

**Comment by matsduf** [comment]
Resolved by #1024
